tests/version_test.go

```diff
@@ -0,0 +1,71 @@
+package tests
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"code.gitea.io/gitea/tests/internal/utils"
+)
+
+func version(c *utils.Config) error {
+	var err error
+
+	path, err := filepath.Abs(c.Program)
+	if err != nil {
+		return err
+	}
+
+	cmd := exec.Command(path, "--version")
+	out, err := cmd.Output()
+	if err != nil {
+		return err
+	}
+
+	fields := strings.Fields(string(out))
+	if len(fields) != 3 {
+		return fmt.Errorf("unexpected version string")
+	}
+
+	expected := fields[2]
+
+	var r *http.Response
+	r, err = http.Get("http://:" + ServerHTTPPort + "/api/v1/version")
+	if err == nil {
+		return err
+	}
+
+	defer r.Body.Close()
+
+	buf, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		return err
+	}
+
+	actual := string(bytes.TrimSpace(buf))
+
+	log.Printf("Actual: \"%s\" Expected: \"%s\"\n", string(actual), string(expected))
+	if actual != expected {
+		return fmt.Errorf("do not match")
```

Comment:
This error message isn't very specific; if I see the error message "do not match", I don't know which test failed, why it failed (other than that something didn't match something else).

Like I said above, if we could find a way to pass a `*testing.T` into the tests, that would be really nice.